vcr/api/v2/registry.go

```
@@ -61,110 +61,94 @@ func (w *Wrapper) SearchVCs(ctx echo.Context) error {
-		Subject      string `json:"subject"`
-		Resources    struct {
-			Path string `json:"path"`
-		} `json:"resources"`
-	}
+func flattenSlice(expanded []interface{}, currentPath []string) []vcr.SearchTerm {
+	result := make([]vcr.SearchTerm, 0)
 
-	var subjectsQuery []credentialSubject
-	if err := vcQuery.UnmarshalCredentialSubject(&subjectsQuery); err != nil {
-		return core.InvalidInputError("failed to unmarshall VC subject: %w", err)
+	for _, sub := range expanded {
+		result = append(result, flatten(sub, currentPath)...)
 	}
 
-	// convert to query
-	query, err := w.VCR.Registry().QueryFor(concept.AuthorizationConcept)
-	if err != nil {
-		return err
-	}
-	if vcQuery.Issuer.String() != "" {
-		query.AddClause(concept.Eq("issuer", vcQuery.Issuer.String()))
-	}
-	if len(subjectsQuery) > 0 {
-		subjectQuery := subjectsQuery[0]
-		if subjectQuery.ID != "" {
-			query.AddClause(concept.Eq("credentialSubject.id", subjectQuery.ID))
-		}
-		if subjectQuery.PurposeOfUse != "" {
-			query.AddClause(concept.Eq("credentialSubject.purposeOfUse", subjectQuery.PurposeOfUse))
-		}
-		if subjectQuery.Subject != "" {
-			query.AddClause(concept.Eq("credentialSubject.subject", subjectQuery.Subject))
-		}
-		if subjectQuery.Resources.Path != "" {
-			query.AddClause(concept.Eq("credentialSubject.resources.#.path", subjectQuery.Resources.Path))
-		}
-	}
+	return result
+}
 
-	results, err := w.VCR.SearchLegacy(ctx.Request().Context(), query, allowUntrusted, nil)
-	if err != nil {
-		return err
+func flattenMap(expanded map[string]interface{}, currentPath []string) []vcr.SearchTerm {
+	// JSON-LD in expanded form either has @value, @id, @list or objects
+
+	results := make([]vcr.SearchTerm, 0)
+
+	for k, v := range expanded {
+		switch k {
+		case "@id":
+			fallthrough
+		case "@value":
+			results = append(results, vcr.SearchTerm{
+				IRIPath: currentPath,
+				Value:   v,
+			})
+		case "@list":
+			// TODO: not supported... This would require OR type queries
```

maybe return an error indicating the query isn't supported? Otherwise it might lead to the querier falsely assuming there are no VCs with the given parameters?

---

actually added every item in the list as required match. This would still be according to spec (all must match)